x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/TransportEqlSearchAction.java

```diff
@@ -134,7 +138,7 @@ public EqlSearchTask createTask(
 
     @Override
     public void execute(EqlSearchRequest request, EqlSearchTask task, ActionListener<EqlSearchResponse> listener) {
-        operation(planExecutor, task, request, username(securityContext), transportService, clusterService, listener);
+        retryAction.executeWithRetry(task, request, listener);
```

Comment:
I guess this would be tighter, but also not far from existing current implementation? And maybe not as close to what [this comment](https://github.com/elastic/elasticsearch/pull/80384#discussion_r744632567) hints to, compared to proposed implementation.
(Also curious how that hinted implementation would look like).